R/provider-ollama.R

```diff
@@ -90,3 +106,26 @@ has_ollama <- function(base_url = "http://localhost:11434") {
     httr2_error = function(cnd) FALSE
   )
 }
+
+method(as_json, list(ProviderOllama, TypeObject)) <- function(provider, x) {
+  if (x@additional_properties) {
+    cli::cli_abort("{.arg .additional_properties} not supported for Ollama.")
+  }
+
+  names <- names2(x@properties)
+
+  properties <- lapply(x@properties, as_json, provider = provider)
+
+  # Unlike OpenAI, Ollama uses the `required` field to list required tool args
+  is_required <- vapply(x@properties, function(t) t@required, logical(1))
```

Comment:
You might take a quick look at some of the other provider methods for `as_json(, TypeObject)` just in case someone else uses the same pattern (which looks vaguely familiar to me)

Reply:
You're right, it's basically the same as for `ProviderGroq`. I refactored to match the code style. b0946386b0dd39a46a66dc42979e06563a217613